src/dvclive/system_metrics.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from typing import Dict, Union
+
+import psutil
+from statistics import mean
+from threading import Event, Thread
+
+from .utils import append_dict
+
+logger = logging.getLogger("dvclive")
+MEGABYTES_DIVIDER = 1024.0**2
+GIGABYTES_DIVIDER = 1024.0**3
+
+MINIMUM_CPU_USAGE_TO_BE_ACTIVE = 30
+
+
+class CPUMetricsCallback:
+    def __init__(
+        self,
+        duration: Union[int, float] = 5,
+        interval: Union[int, float] = 1.0,
+        plot: bool = True,
+    ):
+        self.duration = duration
+        self.interval = interval
+        self.plot = plot
+        self._no_plot_metrics = ["system/cpu/count", "system/cpu/ram_total_GB"]
+
+    def __call__(self, live):
+        self._live = live
+        self._shutdown_event = Event()
+        Thread(
+            target=self.monitoring_loop,
+        ).start()
+
+    def monitoring_loop(self):
+        while not self._shutdown_event.is_set():
+            self._cpu_metrics = {}
+            for _ in range(int(self.duration // self.interval)):
+                last_cpu_metrics = {}
+                try:
+                    last_cpu_metrics = get_cpus_metrics()
+                except psutil.Error:
+                    logger.exception("Failed to monitor CPU metrics:")
+                self._cpu_metrics = append_dict(self._cpu_metrics, last_cpu_metrics)
```

Comment:
pls, check funcy lib ... just in case .. potentially zip or something like that can be used? (so that we don't have additional helpers or we can simplify them)

Reply:
Yeap, problem resolved with `funcy`, thanks! I didn't want to introduce another dependency to the project, but it was already used.